### ISSUE TITLE
Failed to read configuration snapshot from plan file: config snapshot does not have manifest file.

### ISSUE THREAD

**Issue by megan07** [hashicorp/terraform#28266]
### Terraform Version

```
0.15.0-beta2, 0.15.0-rc1
```

### Terraform Configuration Files

I had 12 failing tests with this error. All steps that failed were `Destroy: true`. I'm not sure if this would be something for the SDK or core (we're currently using SDK version `2.4.4`, but I tested it with `2.5.0` and had the same error)

```terraform
resource "google_storage_bucket" "bucket" {
  name = "%s"

  force_destroy = true

  retention_policy {
    retention_period = 3600
  }
}
```

### Debug Output

```
          provider_test.go:267: Step 2/3, expected an error with pattern, no match on: Error running destroy: exit status 1
              
              Error: Invalid plan file
              
              Failed to read configuration snapshot from plan file: config snapshot does
              not have manifest file.
          testing_new.go:63: Error running post-test destroy, there may be dangling resources: exit status 1
              
              Error: could not delete non-empty bucket due to error when deleting contents: googleapi: Error 403: Object 'tf-test-acl-bucket-5872951205364466092/bucketDestroyTestFile' is subject to bucket's retention policy and cannot be deleted, overwritten or archived until 2021-04-01T02:21:08.323117-07:00, forbidden
              
```

### Expected Behavior

The test should pass.

### Actual Behavior

The test is not passing.

**Comment by alisdair** [comment]
Some more details on a repro follow. Config:

```hcl
resource "google_storage_bucket" "bucket" {
  name = "paddy-test-123"
  force_destroy = true
  retention_policy {
    retention_period = 3600
  }
}
```

1. `terraform init`
2. `terraform apply -auto-approve`
3. `terraform plan -no-color -input=false -detailed-exitcode -lock-timeout=0s -out=tfplan -lock=true -parallelism=10 -refresh=false`
4. `terraform apply tfplan`

I haven't personally been able to reproduce this yet due to Google Cloud credentials problems.

**Comment by alisdair** [comment]
Having fixed my Google Cloud credentials, I _still_ can't reproduce this. @paddycarver has been able to consistently reproduce with these steps, though. I think the failure might be masked in my environment by the panic in #28271, so we can reinvestigate once that's fixed.

**Comment by alisdair** [comment]
This seems to have been resolved by #28272. The issue seems to be that the panic in #28271 results in a partially-written plan file, and applying it fails with the error message above.
